deep_tempering/callbacks.py

```diff
@@ -31,7 +31,8 @@ def configure_callbacks(callbacks,
                         mode=ModeKeys.TRAIN,
                         exchange_data=None,
                         swap_step=None,
-                        burn_in=None):
+                        burn_in=None,
+                        **kwargs):
```

Comment:
1. The `kwargs` should be passed directly to the `ExchangeCallback`.  It is not good idea to carry kwargs all over the place (passing to each func - `fit()`. By passing kwargs to the `ExchangeCallback` user chooses desired behaviour. For example, if you want to define a callback with non-default value of coefficient, you define:
```python
clbk = MetropolisExchangeCallback(exchange_data, swap_step, burn_in, coeff=2.)
model.fit(..., callbacks=[clbk]
```
Otherwise, the default behaviour is used. This way there is no need to modify `fit()` arguments for each new variable we wish to use.

You could also create your own callbacks (by subclassing `BaseExchangeCallback`), it doesn't have to be a part of this package, especially on the stage of the prototyping. If at the end we will see that the callbacks work well and we could use them as default behaviour we could add them here as well. 

2. `swap_step` and `burn_in` is also better to remove (but let's keep it as is for now).